docs(community): Deprecation guidelines (#1857)
spinnaker/spinnaker.github.io: community/contributing/back-end-code.md

Exception types you create should extend `RuntimeException` (directly or indirectly).
Please try to include descriptive information in exception messages, especially for errors that will be surfaced to the user.

The package `kork-exceptions` includes some standard base exception types that you are encouraged to use directly or extend as needed.

## Deprecations

Deprecating old, unused or high-debt code is highly encouraged!
When deprecating code, you MUST include the `@Deprecation` annotation, _along with supporting documentation_:

> 1. Why is this code deprecated? This should preferrably be a link to a Github Issue (label it with `no-lifecycle`).

Reviewer: ```suggestion
1. Why is this code deprecated? This should be a link to a Github Issue with the `no-lifecycle` label.
```

1. What is this code being replaced by?

> It is insufficient to simply annotate code as deprecated without additional context: Deprecations will be rejected without sufficient context.

Reviewer: ```suggestion
Do not annotate code as deprecated without additional context. Deprecations without sufficient context will be rejected.
